### ISSUE TITLE
Suggestion: Load DataContext without making connections to datasources.

### ISSUE THREAD

**Issue by kkwanyang** [fivetran/great_expectations#1201]
That title is probably incorrect but I can't seem to find the 1 liner to describe this.

The first step in using GE (after initial setup) is to basically load the DataContext. The datacontext contains all the GE configuration including datasources, loading the datacontext invokes connections to these datasources. If per chance one of these datasources is configured incorrectly, you are unable to load the datacontext, thus unable to use GE. 

In my head it seems like a pattern of determining the suite that is being used before attempting to use connections is a more natural way to  go about this.

TLDR: loading the datacontext involves making connection attempts, and that could fail if a datasource is broken, even though the suite you are trying to create/modify/validate does not depend on that datasource.

**Comment by jcampbell** [comment]
I think this is a good idea -- interested in crafting a PR? I think it would be fairly straightforward, moving some initialization code to happen when get_datasource is called (and ensuring that the CLI is updated to call get_datasource in addition to trying to instantiate a context to know that the config is valid).
